--- configure.ac~	2009-02-22 23:57:17.000000000 +0100
+++ configure.ac	2009-03-14 14:08:55.000000000 +0100
@@ -163,7 +163,7 @@
 if test "x$ac_arg_use_tcl" != "xno"; then
   ac_use_tcl=no
   temp_LIBS="$LIBS"
-  LIBS=""
+  LIBS="-ltcl"
   LIBTCL_LIBS=""
   if test "x$ac_arg_use_autopackage" = xyes; then
     AC_CHECK_HEADERS([tcl8.4/tcl.h tcl.h], [ac_use_tcl=yes
--- src/util_tcl.c.tcl	2009-03-14 14:23:14.000000000 +0100
+++ src/util_tcl.c	2009-03-14 14:41:32.000000000 +0100
@@ -123,9 +123,11 @@
 }
 
 #define TCL_CHECK_PARMS(n) {if (argc < n + 2) { \
-        snprintf (interp->result, TCL_RESULT_SIZE, \
+	char tmp[TCL_RESULT_SIZE]; \
+        snprintf (tmp, TCL_RESULT_SIZE, \
             i18n (2361, "Wrong number of arguments for command '%s %s'. Expected %d.\n"), \
                   argv[0], argv[1], n); \
+	Tcl_SetResult(interp, tmp, TCL_VOLATILE); \
         return TCL_ERROR; \
         } \
     }
@@ -266,8 +268,10 @@
         return TCL_command_help (cd, interp, argc, argv);
     else
     {
-        snprintf (interp->result, TCL_RESULT_SIZE, 
+	char tmp[TCL_RESULT_SIZE];
+        snprintf (tmp, TCL_RESULT_SIZE, 
             i18n (2365, "unknown command: %s"), argv[1]);
+	Tcl_SetResult(interp, tmp, TCL_VOLATILE);
         return TCL_ERROR;
     }
     return TCL_OK;
