--- configure.ac.gnutls	2010-09-22 15:07:03.000000000 +0200
+++ configure.ac	2010-09-22 15:07:30.000000000 +0200
@@ -96,9 +96,6 @@
     ac_use_ssl=no
     if test "x$ac_arg_use_ssl" != "xopenssl"; then
       PKG_CHECK_MODULES(LIBGNUTLS, gnutls > 2, ac_use_ssl=gnutls, ac_use_ssl=no)
-      if test "x$ac_use_ssl" = xno; then
-        AM_PATH_LIBGNUTLS(0.8.8, ac_use_ssl=gnutls) 
-      fi
       AM_PATH_LIBGCRYPT(, , ac_use_ssl=no)
       AC_CHECK_HEADERS([gcrypt.h], , ac_use_ssl=no)
       if test "x$ac_use_ssl" = "xgnutls"; then
